### PR TITLE
Allow the function that returns the time to be replaced

### PR DESCRIPTION
**Pull request by mopemope**
Enabled to replace the function that returns the time.
Allows users to customize the time zone and the time format.

### CODE REVIEW

**Comment by Fs02** [comment]
Hi, may I know the use case for this?

even if you override the timezone for value passed from REL to driver, this function won't be able to handle value retrieved from database, it's still returning as is.
better approach maybe just to configure timezone used to connect via database driver, and setting `time.Local`. example:
https://github.com/go-rel/rel/blob/9969ceac156830eae893e6be42918442a617e1d1/adapter/postgres/postgres_test.go#L19-L30

**Comment by abby-ma2** [comment]
Thanks for the reply.
Rather, my goal is to change the time format.
I want to put values in milliseconds in created_at and so on.

**Comment by abby-ma2** [comment]
Thanks for the reply.
Rather, my goal is to change the time format.
I want to put values in milliseconds in created_at and so on.

**Comment by mopemope** [comment]
Thanks for the reply.
Rather, my goal is to change the time format.
I want to put values in milliseconds in created_at and so on.

**Comment by Fs02** [comment]
I see, so you mean you want to store timestamp as integer? instead of using `createdAt time.Time` you want to use `createdAt int`?

**Comment by Fs02** [comment]
ah sorry, I understood, you want the time precision to be millisecond, not second right? 🤔

**Comment by mopemope** [comment]
Yes. That's right.

File: structset.go [hunk 1]
```diff
@@ -6,9 +6,12 @@ import (
 )
 
 var (
-	now = time.Now
+	NowFunc NowTimeFunc = defaultNow
```
Comment (Fs02) [inline structset.go:9]: how about `Now` only? it have the same name as `time.Now`

File: structset.go [hunk 2]
```diff
@@ -6,9 +6,12 @@ import (
...
 )
 
+// NowTimeFunc is the type of function that returns the current time.
+type NowTimeFunc func() time.Time
```
Comment (Fs02) [inline structset.go:13]: I think this can be shortened to `NowFunc` only as well

File: structset.go [hunk 3]
```diff
@@ -131,3 +134,7 @@ func newStructset(doc *Document, skipZero bool) Structset {
 func NewStructset(record interface{}, skipZero bool) Structset {
 	return newStructset(NewDocument(record), skipZero)
 }
+
+func defaultNow() time.Time {
```
Comment (Fs02) [inline structset.go:138]: can we inline this function in variable declaration?

File: repository_test.go [hunk 4]
```diff
@@ -11,11 +11,14 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+var now NowTimeFunc
```
Comment (Fs02) [inline repository_test.go:14]: looks like unused?

Comment (mopemope) [reply repository_test.go:14]: It is used in many tests.
But I will modify it to use only `Now`.

File: repository_test.go [hunk 5]
```diff
@@ -12,8 +12,8 @@ import (
 )
 
 func init() {
-	t := now().Truncate(time.Second)
-	now = func() time.Time {
+	t := time.Now().Truncate(time.Second)
```
Comment (Fs02) [inline repository_test.go:15]: this should fix the coverage I think
```suggestion
	t := Now()
```

**Comment by Fs02** [comment]
Thank you 🎉

**Review by Fs02 — APPROVED** [review]

**Pull request merged**